src/components/sidebarcom.jsx

```diff
@@ -1,133 +1,185 @@
+  //   let counter = 0;
+  //   const [compo, setCompo] = useState(
+  //     <CodeMirror
+  //       options={{ mode: "javascript", lineNumbers: true }}
+  //       value="console.log('hello world')"
+  //       height="80vh"
+  //       width="170vh"
+  //       theme={theme}
+  //       onChange={(editor, data, value) => {
+  //         {
+  //           setextin(textin + editor);
+  //         }
+  //       }}
+  //     />
+  //   );
+  //   let arrary = [{ counter: { key1: iconstate, key2: compo } }];
+
+  if (clickblob === true) {
+    var blob = new Blob([textin], { type: "text/plain;charset=utf-8" });
     FileSaver.saveAs(blob, "hello world.txt");
     changestateblob(false);
-}
-// const onChange = () => {
-//     settextingchan(textingchan + textin);
-// }
-// console.log("Chaning the text...")
-// console.log(textingchan);
+  }
+  // const onChange = () => {
+  //     settextingchan(textingchan + textin);
+  // }
+  // console.log("Chaning the text...")
+  // console.log(textingchan);
 
+  // writeFileP(`output.txt`, textin, (err, data) => {
+  //     console.log(err || data)});
+  // const [compo , setCompo] = useState(<CodeMirror />);
 
-// writeFileP(`output.txt`, textin, (err, data) => {
-//     console.log(err || data)});
-// const [compo , setCompo] = useState(<CodeMirror />);
+  //   let varible;
 
-let varible;
-
-const changingfunc = (e) => {
+  const changingfunc = (e) => {
     const value = e.target.value;
     //console.log(value);
     let array = value.split(".");
     console.log(array[1]);
-    if(array[1] == "txt"){
-        seticonstate(<AiOutlineFileText className="w-5 h-5 absolute ml-3 pointer-events-none"/>)
-    }
-    else if(array[1] === "py" || array[1] === "python"){
-        seticonstate(<FaPython className="w-5 h-5 absolute ml-3 pointer-events-none" />)
+    if (array[1] === "txt") {
+      seticonstate(
+        <AiOutlineFileText className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
+    } else if (array[1] === "py" || array[1] === "python") {
+      seticonstate(
+        <FaPython className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
+    } else if (array[1] === "js" || array[1] === "javascript") {
+      seticonstate(
+        <DiJavascript1 className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
+    } else if (array[1] === undefined) {
+      seticonstate(
+        <GiBrokenHeart className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
+    } else if (array[1] === "rs" || array[1] === "rust") {
+      seticonstate(
+        <DiRust className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
+    } else if (array[1] === "c#") {
+      seticonstate(
+        <SiCsharp className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
+    } else {
+      seticonstate(
+        <AiFillHeart className="w-5 h-5 absolute ml-3 pointer-events-none" />
+      );
     }
-    else if(array[1] === "js" || array[1] === "javascript"){
```

Comment:
need to understand why did u remove this and add that , is it for alignment ?

Reply:
I haven't removed that, check clearly I've just added === instead of ==